src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -0,0 +1,469 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+using System;
+using System.Threading.Tasks;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Management;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.Admin;
+using static Microsoft.SqlTools.ServiceLayer.Admin.AzureSqlDbHelper;
+using System.Globalization;
+using System.Xml;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// Login object type handler
+    /// </summary>
+    public class DatabaseHandler : ObjectTypeHandler<DatabaseInfo, DatabaseViewContext>
+    {
+        private const int minimumVersionForWritableCollation = 8;
+        private const int minimumVersionForRecoveryModel = 8;
+        private const string DefaultValue = "<default>";
+
+        /// <summary>
+        /// Set of valid compatibility levels and their display strings
+        /// </summary>
+        private Dictionary<CompatibilityLevel, string> compatLevels = new Dictionary<CompatibilityLevel, string>()
+        {
+            {CompatibilityLevel.Version70, "SQL Server 7.0 (70)"},
+            {CompatibilityLevel.Version80, "SQL Server 2000 (80)"},
+            {CompatibilityLevel.Version90, "SQL Server 2005 (90)"},
+            {CompatibilityLevel.Version100, "SQL Server 2008 (100)"},
+            {CompatibilityLevel.Version110, "SQL Server 2012 (110)"},
+            {CompatibilityLevel.Version120, "SQL Server 2014 (120)"},
+            {CompatibilityLevel.Version130, "SQL Server 2016 (130)"},
+            {CompatibilityLevel.Version140, "SQL Server 2017 (140)"},
+            {CompatibilityLevel.Version150, "SQL Server 2019 (150)"},
+            {CompatibilityLevel.Version160, "SQL Server 2022 (160)"},
+        };
+
+        public DatabaseHandler(ConnectionService connectionService) : base(connectionService)
+        {
+        }
+
+        public override bool CanHandleType(SqlObjectType objectType)
+        {
+            return objectType == SqlObjectType.Database;
+        }
+
+        public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRequestParams requestParams)
+        {
+            // open a connection for running the database dialog and associated task
+            ConnectionInfo originalConnInfo;
```

Comment:
There's a lot of code in here, in the interest of being able to accurately review it in a timely manner is there any way you could just temporarily delete the stuff you don't need? I would rather avoid shoving a bunch of unreviewed stuff in here, even if it isn't actively being used.

Reply:
Removed the extra stuff